### PR TITLE
SY-1657 improve datetime input

### PR DESCRIPTION
**Pull request by emilbon99**
# Issue Pull Request

## Key Information

- **Linear Issue**: [SY-####](https://linear.app/synnax/issue/)

## Basic Readiness

- [ ] I have performed a self-review of my code.
- [ ] I have added relevant tests to cover the changes to CI.
- [ ] I have added needed QA steps to the [release candidate](/synnaxlabs/synnax/blob/main/.github/PULL_REQUEST_TEMPLATE/rc.md) template that cover these changes.
- [ ] I have updated in-code documentation to reflect the changes.
- [ ] I have updated user-facing documentation to reflect the changes.

### Data Structures

I have ensured that previous versions of stored data structures are properly migrated to new formats in the following projects:

- [ ] Server
- [ ] Console

### API Changes

The following projects have backwards-compatible APIs:

- [ ] Python Client
- [ ] Server
- [ ] TypeScript Client

### CODE REVIEW

File: pluto/src/align/Space.tsx [hunk 1]
```diff
@@ -111,7 +111,7 @@ const CoreSpace = <E extends SpaceElementType>(
   else if (typeof size === "number") gap = `${size}rem`;
 
   style = {
-    gap,
+    gap: `${gap} ${gap}`,
```
Comment (pjdotson) [inline pluto/src/align/Space.tsx:114]: Why do you need to change this?

File: pluto/src/index.css [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    *:not(:is(.pluto-text, input, label)) {
+        user-select: none !important;
+        -webkit-user-select: none !important;
+    }
+
+    box-sizing: border-box;
+}
+
+/*
+ * Copyright 2024 Synnax Labs, Inc.
```
Comment (pjdotson) [inline pluto/src/index.css:42]: Don't need copyright twice

File: pluto/src/dropdown/Dropdown.css [hunk 3]
```diff
@@ -38,11 +38,14 @@
     background: var(--pluto-gray-l3-35);
 
     .pluto-dropdown__dialog {
+        /* contain: layout size style; */
```
Comment (pjdotson) [inline pluto/src/dropdown/Dropdown.css:41]: Commented code

File: pluto/src/nav/Bar.css [hunk 4]
```diff
@@ -12,7 +12,7 @@
 .pluto-navbar {
     position: relative;
     flex-shrink: 0;
-    background: var(--pluto-gray-l0);
+    /* background: var(--pluto-gray-l0); */
```
Comment (pjdotson) [inline pluto/src/nav/Bar.css:15]: Commented code

File: pluto/src/text/Link.css [hunk 5]
```diff
@@ -18,7 +18,3 @@
 .pluto-text-link:hover {
     text-decoration: underline;
 }
-
-.pluto-text-link__text {
-    color: var(--pluto-primary-p1);
```
Comment (pjdotson) [inline pluto/src/text/Link.css:23]: Why was this removed?

File: pluto/src/input/Input.css [hunk 6]
```diff
@@ -19,6 +19,23 @@
...
+    justify-content: center;
+    align-items: center;
+    top: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+    background: var(--pluto-gray-l2);
+    z-index: 2;
+    * {
+        /* color: var(--pluto-gray-l9) !important; */
```
Comment (pjdotson) [inline pluto/src/input/Input.css:34]: Commented out code

File: console/src/range/overview/Overview.tsx [hunk 7]
```diff
@@ -10,6 +10,8 @@
 import "@/range/overview/Overview.css";
 
 import { Align, Divider } from "@synnaxlabs/pluto";
+import nlp from "compromise";
+import nlpDates from "compromise-dates";
```
Comment (pjdotson) [inline console/src/range/overview/Overview.tsx:14]: What does importing here do? And better names please

**Review by pjdotson — CHANGES_REQUESTED** [review]
On date selector:
- leap days don't exist for February
- If I click on August 31, and hit the next month, it skips me over to october 31, skipping september
- If I'm on May 31st and I click the previous month, I go to May 1
- What If we didn't offer millisecond-level precision in the date selector? I have a tough time imagining when someone wants that exact level of precision but didn't use the python script or created a range from selecting on a plot. This could clear up some of the space and make the date + times nicer to look at (removes 4 characters from each shown date time) 

Screenshot 2024-12-14 at 9 29 41 AM

Would also be good to remove the blue primary color from the hour / minute / time list in light mode

File: console/src/range/overview/Overview.tsx (hunk 7 above)
Comment (emilbon99) [reply console/src/range/overview/Overview.tsx:14]: These are useless imports. Also `nlp` means natural language processing - seems like a common enough acronym to use the abbreviation.

Comment (emilbon99) [reply console/src/range/overview/Overview.tsx:14]: I removed but they still exist in the date picker
